src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
-
+#endif
+
+void func_80A546DC(EnHeishi2* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    Math_SmoothScaleMaxMinS(&thisx->shape.rot.z, 200, 5, this->unk_2E4, 0);
+    Math_SmoothScaleMaxF(&this->unk_2E4, 3000.0f, 1.0f, 500.0f);
+    thisx->posRot.rot.z = thisx->shape.rot.z;
+    if (thisx->shape.rot.z > 0) {
+        Actor_Kill(thisx);
+    }
+}
+
+#ifdef NON_MATCHING
+// can't get this to match, i think it's mostly functionally equivalent
+void func_80A5475C(EnHeishi2* this, GlobalContext* globalCtx) {
+    Actor* thisx;
+    s16 angleDiff;
+    s16 newAngleDiff;
+    s16 phi_v0;
+
+    thisx = &this->actor;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (Text_GetFaceReaction(globalCtx, 5) != 0) {
+        if (this->unk_30B == 0) {
+            if (this->initParams == 2) {
+                this->actionFunc = func_80A53278;
+                return;
+            }
+            if (this->initParams == ((((u16)5) & 0xFFu))) {
+                this->actionFunc = func_80A5399C;
+                return;
+            }
+        }
+    }
+
+    else {
+        if (this->unk_30B != 0) {
+            if (this->initParams == 2) {
+                this->actionFunc = func_80A53278;
+                return;
+            }
+            if (5 == this->initParams) {
+                this->actionFunc = func_80A5399C;
+                return;
+            }
+        }
+    }
+    if ((func_8002F194(thisx, globalCtx) != 0)) {
+        if (this->initParams == 2) {
+            if (this->unk_30E == 1) {
+                this->actionFunc = func_80A5344C;
+                return;
+            }
+            this->actionFunc = func_80A53278;
+            return;
+        }
+
+        if (this->initParams == 5) {
+            phi_v0 = this->unk_300;
+
+            if (phi_v0 == 6) {
+                this->actionFunc = func_80A5399C;
+                phi_v0 = this->unk_300;
+                do {
+                    phi_v0 = this->unk_300;
+                } while (0);
+            }
+
+            if (phi_v0 == 5) {
+                this->actionFunc = func_80A54954;
+                phi_v0 = this->unk_300;
+                do {
+                    phi_v0 = this->unk_300;
+                } while (0);
+            }
+
+            if (phi_v0 != 4) {
+                return;
+            }
+            this->unk_309 = 1;
+            func_80078884(0x4807);
+            this->actionFunc = func_80A540C0;
+            return;
+        }
+
+        phi_v0 = this->initParams;
+    } else {
+        if (this->initParams == 2) {
+            angleDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
+            if (angleDiff >= 0) {
+                newAngleDiff = angleDiff;
+            } else {
+                newAngleDiff = -angleDiff;
+            }
+            if (!(120.0f < this->actor.xzDistanceFromLink)) {
+                if (newAngleDiff < 0x4300) {
```

Comment:
this pattern is ABS() again

Reply:
you should still be able to use the abs macro here. i would put it even if this function is commented out, just so its there for whoever continues this